owasp-top10-2016-mobile/m2/cool_games/README.md

```diff
@@ -0,0 +1,178 @@
+</p>
+
+### 🔥
+
+If an attacker came into posession of the device and used the [Android Debug Bridge](https://developer.android.com/studio/command-line/adb) tool to communicate with it through a Unix shell, he could inspect how the app behaves and what it logs.
+
+To begin, after connecting the device through an USB port, it is possible to list connected devices with the following command:
+
+```sh
+adb devices
+```
+
+<p align="center">
+    <img src="images/adb_devices.png"/>
+</p>
+
+Note: This attack narrative is being written using an emulator as example, thus the `emulator` showing up as a device.
+
+Now that the device is properly connected, we can start a shell instance inside it through the command:
+
+```sh
+adb shell
+```
+
+<p align="center">
+    <img src="images/adb_shell.png"/>
+</p>
+
+In order to verify the app's log, we need to first know it's `package name`, which can be found in the `AndroidManifest.xml` file located at:
+
+```sh
+secDevLabs/owasp-top10-2016-mobile/m2/cool_games/mobile/android/app/src/profile/AndroidManifest.xml
+```
+
+<p align="center">
+    <img src="images/package_name.png"/>
+</p>
+
+Now that we know `com.secdevlabs.cool.games` is the app's package name, we can use it to filter logs through the `logcat` command, available through the `adb shell`, as shown below:
+
+```sh
+logcat com.secdevlabs.cool.games
+```
+
+<p align="center">
+    <img src="images/logcat.png"/>
+</p>
+
+Having access to the app's log, it's now a matter of searching through it. Through a quick hint of filtering for `"password"`, we can see log in attempts logging passwords, as shown by the following image:
+
+<p align="center">
+    <img src="images/logged_password.png"/>
+</p>
+
+### 👀
+
+#### Lack of view filling when switching apps allows for possible sensitive information leakage
+
+Both on Android or iOS there's an option to switch between recently opened apps by clicking on the app's screen and getting right back into the action. But in order for the system to show the screen of a recent used app, a screenshot is taken on the exact moment you click to switch and send it to the background, as shown by the following images:
+
+<p align="center">
+    <img src="images/recent_app_android.png"/>
+</p>
+
+### 🔥
+
+If an attacker were to obtain the device and could manage to either root it or jump to the app's process, it would be possible to visualize the screenshot taken in the app's sandbox directory.
+
+In another scenario, if the device was obtained shortly after the user was using it, there's a possibility that an attacker could find sensitive information through the recent app's view, as shown by the image below:
+
+<p align="center">
+    <img src="images/recent_app_password.png"/>
+</p>
+
+## Secure this app
+
+How would you mitigate this vulnerability? After your changes, an attacker should not be able to:
+
+* View sensitive data being logged, as the app no longer logs them.
+* View screen of recently opened apps.
```

Comment:
We should make it explicit here that we're talking about the cool games app screen since using plural here may mislead users into thinking that we're talking about system configuration and not app configuration.

Reply:
Thanks for the feedback, @joserenatosilva! I agree!